engine.fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/SearchParamFhirRetrieveProvider.java

```diff
@@ -64,216 +53,19 @@ public int getMaxCodesPerQuery() {
-                codeParams.addOr(new TokenParam(s));
-            }
-
-        }
-
-        if (codeParams != null) {
-            codeParamsList.add(codeParams);
-        }
-
-        return codeParamsList;
-    }
-
-    protected List<SearchParameterMap> setupQueries(String context, String contextPath, Object contextValue,
-            String dataType, String templateId, String codePath, Iterable<Code> codes, String valueSet, String datePath,
-            String dateLowPath, String dateHighPath, Interval dateRange) {
-
-        Pair<String, IQueryParameterType> templateParam = this.getTemplateParam(dataType, templateId);
-        Pair<String, IQueryParameterType> contextParam = this.getContextParam(dataType, context, contextPath,
-                contextValue);
-        Pair<String, DateRangeParam> dateRangeParam = this.getDateRangeParam(dataType, datePath, dateLowPath,
-                dateHighPath, dateRange);
-        Pair<String, List<TokenOrListParam>> codeParams = this.getCodeParams(dataType, codePath, codes, valueSet);
-
-        // In the case we filtered to a valueSet without codes, there are no possible
-        // results.
-        if (valueSet != null && (codeParams == null || codeParams.getValue().isEmpty())) {
-            return Collections.emptyList();
-        }
-
-        return this.innerSetupQueries(templateParam, contextParam, dateRangeParam, codeParams);
-    }
-
-    protected List<SearchParameterMap> innerSetupQueries(Pair<String, IQueryParameterType> templateParam,
-            Pair<String, IQueryParameterType> contextParam, Pair<String, DateRangeParam> dateRangeParam,
-            Pair<String, List<TokenOrListParam>> codeParams) {
-
-        if (codeParams == null || codeParams.getValue() == null || codeParams.getValue().isEmpty()) {
-            return Collections.singletonList(this.getBaseMap(templateParam, contextParam, dateRangeParam));
-        }
-
-        List<SearchParameterMap> maps = new ArrayList<>();
-        for (TokenOrListParam tolp : codeParams.getValue()) {
-            SearchParameterMap base = this.getBaseMap(templateParam, contextParam, dateRangeParam);
-            base.add(codeParams.getKey(), tolp);
-            maps.add(base);
-        }
-
-        return maps;
-    }
-
-    protected SearchParameterMap getBaseMap(Pair<String, IQueryParameterType> templateParam,
-            Pair<String, IQueryParameterType> contextParam, Pair<String, DateRangeParam> dateRangeParam) {
-
-        SearchParameterMap baseMap = new SearchParameterMap();
-        baseMap.setLastUpdated(new DateRangeParam());
-
-        if (this.pageSize != null) {
-            baseMap.setCount(this.pageSize);
-        }
-
-        if (templateParam != null) {
-            baseMap.add(templateParam.getKey(), templateParam.getValue());
-        }
+    @Override
+    public Iterable<Object> retrieve(ModelResolver modelResolver, String context, String contextPath, Object contextValue, String dataType,
```

Comment:
Can't add ModelResolver to the public API here. The engine doesn't know about them, so it can't pass them in.